autotest/ogr/ogr_shape.py

```diff
@@ -1898,13 +1898,15 @@ def test_ogr_shape_48(tmp_vsimem):
     )
     lyr.CreateFeature(feat)
     feat.SetGeometry(
-        ogr.CreateGeometryFromWkt("POLYGON((0 0 2,0 1 2,1 1 2,1 0 2,0 0 2))")
+        ogr.CreateGeometryFromWkt("POLYGON((0 0 2,0 1 1,1 1 2,1 0 2,0 0 3))")
     )
     lyr.SetFeature(feat)
     ds.ExecuteSQL("RECOMPUTE EXTENT ON ogr_shape_48")
-    # FIXME: when we have a GetExtent3D
     extent = lyr.GetExtent()
     assert extent == (0, 1, 0, 1), "did not get expected extent (4)"
+    extent3D = lyr.GetExtent3D()
+    assert lyr.TestCapability(ogr.OLCFastGetExtent)
```

Comment:
shouldn't that be ?
```suggestion
    assert lyr.TestCapability(ogr.OLCFastGetExtent3D)
```

It would be great to also test GetExtent3D() on a 2D only shapefile. Too check that we get NaN in the minZ/maxZ fields